ocaml/xapi-types/ref.mli

```diff
@@ -20,6 +20,10 @@ val make : unit -> 'a t
 
 val null : 'a t
 
+val compare : 'a t -> 'a t -> int
```

Comment:
Looks good, but you don't seem to be using it yet?

Reply:
There are some places where refs are "setified", this lowers the barrier to create sets instead.